core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -675,78 +812,88 @@ impl ForkChoice for HeaviestSubtreeForkChoice {
         bank_forks: &RwLock<BankForks>,
     ) -> (Arc<Bank>, Option<Arc<Bank>>) {
         let r_bank_forks = bank_forks.read().unwrap();
-
         (
-            r_bank_forks.get(self.best_overall_slot()).unwrap().clone(),
+            // BankForks should only contain one valid version of this slot
```

Comment:
yeah only frozen banks are added.

This also means before we dump any block for duplicate repair, we have to make sure to mark it as invalid so that it doesn't trigger this assert.